packages/plugin-hardhat/src/index.ts

```diff
@@ -116,3 +116,15 @@ extendEnvironment(hre => {
     };
   });
 });
+
+extendConfig((config: HardhatConfig) => {
+  for (const compiler of config.solidity.compilers) {
+    if (compiler.settings.outputSelection['*']['*'] === undefined) {
```

Comment:
this will fail if:
- `compiler.settings` is undefined,
- `compiler.settings.outputSelection` is undefined
- `compiler.settings.outputSelection['*']` is undefined is undefined 

we need another way of setting that

Reply:
maybe

```
Object.assign({ settings: {}}, compiler);
Object.assign({ outputSelection: {}}, compiler.settings);
Object.assign({ '*': {}}, compiler.settings.outputSelection);
Object.assign({ '*': {}}, compiler.settings.outputSelection['*']);
```

Reply:
```
const set = (obj, key, value = {}) => (key in obj) || (obj[key] = value);
set(compiler, 'settings');
set(compiler.settings, 'outputSelection');
set(compiler.settings.outputSelection, '*');
set(compiler.settings.outputSelection['*'], '*', []);
```

Reply:
this is marked outdated, but is not resolved